profiles/common/modules/features/nexteuropa_dgt_connector/tmgmt_poetry/tmgmt_poetry.module

```diff
@@ -711,70 +709,420 @@ function tmgmt_poetry_form_tmgmt_entity_ui_translate_form_alter(&$form, &$form_s
+
+      $states = array_values(_tmgmt_workbench_available_states());
+      $state = $node->workbench_moderation['current']->state;
+      $translation_possible = in_array($state, $states, 'strict');
+      // The custom header with for the 'last change' column.
+      $form['languages']['#header'][TMGMT_WORKBENCH_REQUEST_LAST_CHANGE_INDEX] = t('Last change');
+
+      // Setting up an array with language codes based on the available options.
+      $lang_codes = array_keys($form['languages']['#options']);
+
+      foreach ($lang_codes as $lang_code) {
+        // The 'last change' custom column by default is empty.
+        $form['languages']['#options'][$lang_code][TMGMT_WORKBENCH_REQUEST_LAST_CHANGE_INDEX] = '';
+        if (isset($existing_jobs[$lang_code])) {
+          $item = $existing_jobs[$lang_code];
+          $wrapper = entity_metadata_wrapper('tmgmt_job_item', $item);
+
+          $form['languages']['#options'][$lang_code][TMGMT_WORKBENCH_REQUEST_LAST_CHANGE_INDEX]
+            = format_date($wrapper->changed->value(), 'short');
+        }
+      }
 
       if (!empty($existing_jobs)) {
-
-        $translation_possible = TRUE;
-
-        // Check the poetry status of subjobs.
-        foreach ($existing_jobs as $existing_job) {
-          // Check if it's a poetry job.
-          $translator = $existing_job->getTranslator();
-          if ($translator && $translator->plugin != 'poetry') {
-            return;
-          }
-
-          // Check the poetry status.
-          $result = db_select('poetry_status', 'ps')
-            ->fields('ps')
-            ->condition('tjiid', $existing_job->tjiid, '=')
-            ->execute()
-            ->fetchAssoc();
-
-          if (isset($result['status']) && $result['status'] == POETRY_STATUS_MESSAGE_PEN) {
-            $translation_possible = FALSE;
-          }
+        if (_tmgmt_poetry_check_jobs_translatability($existing_jobs)) {
+          _tmgmt_poetry_inject_request_update_elements($form);
+        }
+        else {
+          _tmgmt_poetry_disable_form_elements($form);
+          $form['languages_description'] = [
+            '#type' => 'item',
+            '#prefix' => '<div class="messages warning">',
+            '#markup' => t('Please wait for the translation request to be accepted before further update options.'),
+            '#suffix' => '</div>',
+          ];
         }
 
-        // Force form to allow user actions.
-        $form_state['force_actions'] = TRUE;
+        // If translatable languages are available add the button and other
+        // elements for the additional languages requests functionality.
+        if (_tmgmt_poetry_check_translatability($form_state['tmgmt_cart']['item_id'])
+          && !empty(_tmgmt_poetry_get_translatable_languages($existing_jobs, $form, $form_state))
+        ) {
+          // Getting additional translatable languages.
+          $trans_langs = _tmgmt_poetry_get_translatable_languages($existing_jobs, $form, $form_state);
```

Comment:
I think we should define the trans_langs  variable inside the if statement directly to avoid calling the same function twice.